src/controllers/legacycontrollermappingfilehandler.cpp

```diff
@@ -1,14 +1,63 @@
 #include "controllers/legacycontrollermappingfilehandler.h"
 
+#include <QStringBuilder>
+
 #include "controllers/defs_controllers.h"
 #include "controllers/midi/legacymidicontrollermappingfilehandler.h"
+#include "util/logger.h"
 #include "util/xml.h"
 
 #ifdef __HID__
 #include "controllers/hid/legacyhidcontrollermappingfilehandler.h"
 #endif
 
+#ifdef MIXXX_USE_QML
+QMap<QString, QImage::Format> LegacyControllerMappingFileHandler::kSupportedPixelFormat = {
+        {"RBG", QImage::Format_RGB888},
+        {"RBGA", QImage::Format_RGBA8888},
+        {"RGB565", QImage::Format_RGB16},
+};
+
+QMap<QString, std::endian> LegacyControllerMappingFileHandler::kEndianFormat = {
+        {"big", std::endian::big},
+        {"little", std::endian::little},
+};
+#endif
 namespace {
+const mixxx::Logger kLogger("LegacyControllerMappingFileHandler");
+
+#ifdef MIXXX_USE_QML
+
+/// Find a module directory (QML) in the mapping or system path.
+///
+/// @param mapping The controller mapping the module directory belongs to.
+/// @param dirname The module directory name.
+/// @param systemMappingsPath The system mappings path to use as fallback.
+/// @return Returns a QFileInfo object. If the script was not found in either
+/// of the search directories, the QFileInfo object might point to a
+/// non-existing file.
+QFileInfo findLibraryPath(
+        std::shared_ptr<LegacyControllerMapping> mapping,
+        const QString& dirname,
+        const QDir& systemMappingsPath) {
+    // Always try to load module directory from the mapping's directory first
+    QFileInfo dir = QFileInfo(mapping->dirPath().absoluteFilePath(dirname));
+
+    // If the module directory does not exist, try to find it in the fallback dir
+    if (!dir.isDir()) {
+        dir = QFileInfo(systemMappingsPath.absoluteFilePath(dirname));
+    }
+    return dir;
+}
+
+/// @brief Parse a string that contain a boolean value in human representation
+/// @param value the string containing the boolean setting
+/// @return true for string value "yes", "true" and "1", false otherwise
+bool parseHumanBoolean(const QString& value) {
```

Comment:
I would prefer to enforce a specific allowed term (probably "true") rather than this list.  if we start allowing options, why not french "oui" as well?

Reply:
Fair enough. I have kept "1" too for now, but happy to enforce strictly "true" only if you feel strongly about it.